### PR TITLE
nim2: init 2.0.0

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

Update Nim to 2.0.0. Lots of stuff break.

https://nim-lang.org/blog/2023/08/01/nim-v20-released.html

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/nim-packages/sdl2/default.nix [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
-  version = "2.0.4";
-  src = fetchNimble {
-    inherit (finalAttrs) pname version;
-    hash = "sha256-Vtcj8goI4zZPQs2TbFoBFlcR5UqDtOldaXSH/+/xULk=";
+  version = "2.0.5";
+  src = fetchFromGitHub {
+    owner = "nim-lang";
+    repo = "sdl2";
+    rev = "v${final.version}";
+    hash = "sha256-qDtVSnf+7rTq36WAxgsUZ8XoUk4sKwHyt8EJcY5WP+o=";
```
Comment (erdnaxe) [inline pkgs/development/nim-packages/sdl2/default.nix:10]: I'm getting:
```
error: hash mismatch in fixed-output derivation '/nix/store/1dsj4yrq3yb4cd0xhhavgjvba1y9jf4q-source.drv':
         specified: sha256-qDtVSnf+7rTq36WAxgsUZ8XoUk4sKwHyt8EJcY5WP+o=
            got:    sha256-oUTUWuBphoR0pBMkcJBVDW+dnnF8KK23F7eW3lOLNO4=
```

File: pkgs/development/nim-packages/chroma/default.nix [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
   pname = "chroma";
-  version = "0.2.5";
-  hash = "sha256-6lNHpO2aMorgkaPfo6kRcOs9r5R6T/kislVmkeoulw8=";
-}
+  version = "0.2.7";
+  src = fetchFromGitHub {
+    owner = "treeform";
+    repo = "chroma";
+    rev = final.version;
+    hash = "sha256-EYLzpzmNUwEOEndAwnUXCqpIUMmpinpiZq+P6zO0Kk8=";
```
Comment (erdnaxe) [inline pkgs/development/nim-packages/chroma/default.nix:10]:
```
error: hash mismatch in fixed-output derivation '/nix/store/1zcfjqi9banhc2hrygvvx5560h3hm629-source.drv':
         specified: sha256-EYLzpzmNUwEOEndAwnUXCqpIUMmpinpiZq+P6zO0Kk8=
            got:    sha256-QSl8n60HO56kJ4BKKe/FjUC/cRNxqL2L56FqPwhmJl4=
```

**Review by erdnaxe — COMMENTED** [review]
I am trying to run a nixpkgs-review on this PR to see how much stuff breaks, but `eris-20230722` is never getting build (infinite loop?).

**Comment by Infinidoge** [comment]
Did a quick spot test of Nitter, and it compiles and runs properly as best as I can tell.

File: pkgs/servers/nitter/default.nix [hunk 3]
```diff
@@ -1,12 +1,25 @@
 { lib
+, buildNimPackage
 , fetchFromGitHub
 , nimPackages
```
Comment (erdnaxe) [inline pkgs/servers/nitter/default.nix:4]: nimPackages seems no longer used.

**Comment by erdnaxe** [comment]
spry 0.9.0 fails to build due to:
```
/nix/store/j4kxh4xxlb58dwl99qippw9mwdp5dx1w-snappy-0.1.0/snappy.nim(482, 3) Error: undeclared identifier: 'shallowCopy'
```

**Comment by ehmry** [comment]
> spry 0.9.0 fails to build due to: Error: undeclared identifier: 'shallowCopy'

`nim2Packages.spry` isn't going to build, but `nimPackages.spry` does.

**Pull request merged**